### PR TITLE
Dedupe Hrana hello and repeated hello

### PR DESCRIPTION
**Pull request by shopifyski**
deduplicated handling of hrana hello and repreated hello. Code is fully equivalent to the previous form. No change in logic.

### CODE REVIEW

File: libsql-server/src/hrana/ws/conn.rs [hunk 1]
```diff
@@ -208,12 +208,18 @@ async fn handle_client_msg(conn: &mut Conn, client_msg: proto::ClientMsg) -> Res
 async fn handle_hello_msg(conn: &mut Conn, jwt: Option<String>) -> Result<bool> {
     let hello_res = match conn.session.as_mut() {
         None => {
-            session::handle_initial_hello(&conn.server, conn.version, jwt, conn.namespace.clone())
+            conn.session = session::handle_hello(&conn.server, jwt, conn.namespace.clone())
                 .await
-                .map(|session| conn.session = Some(session))
+                .map(|auth| session::Session::new(auth, conn.version))
+                .map(|s| Some(s))?;
```
Comment (haaawk) [inline libsql-server/src/hrana/ws/conn.rs:214]: Wouldn't the `?` return from `handle_hello_msg` and not execute the `Err(err)` branch in line 231?

File: libsql-server/src/hrana/ws/conn.rs [hunk 2]
```diff
@@ -208,12 +208,18 @@ async fn handle_client_msg(conn: &mut Conn, client_msg: proto::ClientMsg) -> Res
...
+                Ok(())
         }
         Some(session) => {
-            session::handle_repeated_hello(&conn.server, session, jwt, conn.namespace.clone()).await
+            if session.version < Version::Hrana2 {
+                bail!(ProtocolError::NotSupported {what: "Repeated hello message", min_version: Version::Hrana2,})
+            }
+            session.auth = session::handle_hello(&conn.server, jwt, conn.namespace.clone()).await?;
```
Comment (haaawk) [inline libsql-server/src/hrana/ws/conn.rs:221]: Same question about the `?`. Previously this match would never return from `handle_hello_msg`.

File: libsql-server/src/hrana/ws/session.rs [hunk 3]
```diff
@@ -65,60 +78,23 @@ pub enum ResponseError {
...
     let namespace_jwt_key = server
         .namespaces
         .with(namespace.clone(), |ns| ns.jwt_key())
         .await??;
 
-    session.auth = namespace_jwt_key
+    namespace_jwt_key
         .map(Jwt::new)
         .map(Auth::new)
-        .unwrap_or_else(|| server.user_auth_strategy.clone())
```
Comment (haaawk) [inline libsql-server/src/hrana/ws/session.rs:117]: Why is `unwrap_or` better than `unwrap_or_else`?

File: libsql-server/src/hrana/ws/session.rs [hunk 4]
```diff
@@ -15,13 +15,26 @@ use crate::namespace::NamespaceName;
 
 /// Session-level state of an authenticated Hrana connection.
 pub struct Session {
-    auth: Authenticated,
-    version: Version,
+    pub auth: Authenticated,
```
Comment (haaawk) [inline libsql-server/src/hrana/ws/session.rs:18]: I think you could avoid leaking those details by having `handle_initial_hello`, `handle_repeated_hello` that both call private `handle_hello`.

File: libsql-server/src/hrana/ws/conn.rs (hunk 1 above)
Comment (shopifyski) [reply libsql-server/src/hrana/ws/conn.rs]: spot on. I guess inline logic wasnt that smart :)

File: libsql-server/src/hrana/ws/session.rs (hunk 3 above)
Comment (shopifyski) [reply libsql-server/src/hrana/ws/session.rs:117]: great question :)

File: libsql-server/src/hrana/ws/conn.rs [hunk 5]
```diff
@@ -206,15 +206,13 @@ async fn handle_client_msg(conn: &mut Conn, client_msg: proto::ClientMsg) -> Res
...
-                .await
-                .map(|session| conn.session = Some(session))
-        }
-        Some(session) => {
-            session::handle_repeated_hello(&conn.server, session, jwt, conn.namespace.clone()).await
-        }
+        None => auth
+            .map(|auth| session::Session::new(auth, conn.version))
+            .map(|s| conn.session = Some(s)),
+        Some(session) => auth.map(|a| session.update_auth(a)).and_then(|op| op),
```
Comment (haaawk) [inline libsql-server/src/hrana/ws/conn.rs:215]: I'm wondering if the code wouldn't be a bit easier to follow if we structure it like this:
```
let hello_res = match auth {
    Ok(a) => match conn.session.as_mut() {
        None => {
            conn.session = Some(session::Session::new(a, conn.version));
            Ok(())
        }
        Some(session) => session.update_auth(a)
    }
    Err(e) => Err(e)
}
```

**Review by haaawk — COMMENTED** [review]
LGTM. Just one more suggestion to hopefully make the code a bit easier to understand.

**Review by haaawk — APPROVED** [review]

**Review by haaawk — APPROVED** [review]

**Pull request merged**